### ISSUE TITLE
Keras YatConv1D accepts causal padding but cannot execute it

### ISSUE THREAD

**Issue by mlnomadpy** [azettaai/nmn#74]
## Summary

Keras `YatConv1D` accepts `padding="causal"` but cannot execute it.

## Reproduction

On the JAX backend, the call raises:

```text
Unrecognized padding type: expected VALID, SAME, or SAME_LOWER, got causal
```

The implementation bypasses the base `Conv1D` causal pre-padding helper and passes `"causal"` directly to `keras.ops.conv` for the dot product and patch norm.

## Acceptance criteria

- [ ] Pre-pad consistently for both computations, then use valid convolution.
- [ ] Test causality, shapes, stride/dilation validation, and both Keras backends.
